applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_linear_3D2N.cpp

```diff
@@ -51,6 +51,44 @@ CrBeamElementLinear3D2N::Create(IndexType NewId,
 
 CrBeamElementLinear3D2N::~CrBeamElementLinear3D2N() {}
 
+void CrBeamElementLinear3D2N::Initialize(const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    if (this->GetProperties().Has(SEMI_RIGID_NODE_IDS)) {
+
+        KRATOS_ERROR_IF_NOT(this->GetProperties().Has(SEMI_RIGID_ROTATIONAL_STIFFNESS_VECTOR_AXIS_2)) << 
+            "SEMI_RIGID_ROTATIONAL_STIFFNESS_VECTOR_AXIS_2 must be set when SEMI_RIGID_NODE_IDS is used\n";
+        KRATOS_ERROR_IF_NOT(this->GetProperties().Has(SEMI_RIGID_ROTATIONAL_STIFFNESS_VECTOR_AXIS_3)) << 
+            "SEMI_RIGID_ROTATIONAL_STIFFNESS_VECTOR_AXIS_3 must be set when SEMI_RIGID_NODE_IDS is used\n";
+        KRATOS_ERROR_IF_NOT(this->GetProperties()[SEMI_RIGID_NODE_IDS].size() == this->GetProperties()[SEMI_RIGID_ROTATIONAL_STIFFNESS_VECTOR_AXIS_2].size()) << 
+            "SEMI_RIGID_ROTATIONAL_STIFFNESS_VECTOR_AXIS_2 must have equal size as SEMI_RIGID_NODE_IDS\n";
+        KRATOS_ERROR_IF_NOT(this->GetProperties()[SEMI_RIGID_NODE_IDS].size() == this->GetProperties()[SEMI_RIGID_ROTATIONAL_STIFFNESS_VECTOR_AXIS_3].size()) <<
+            "SEMI_RIGID_ROTATIONAL_STIFFNESS_VECTOR_AXIS_3 must have equal size as SEMI_RIGID_NODE_IDS\n";
+
+        // create integer list
+        Vector semi_rigid_node_id_input = this->GetProperties()[SEMI_RIGID_NODE_IDS];

```

Comment:
Here is an example of using the nodal data for specifying the input:
https://github.com/KratosMultiphysics/Kratos/blob/d4a91c0a9fcabda5739e66e281510b34696c3443/applications/StructuralMechanicsApplication/custom_elements/spring_damper_element.cpp#L468

This way you can be sure that the correct nodes have the input
Additionally, for large number of hinges it will it overcrowd your materials.json